[BUG] 루트 엔드포인트에서 NameError 발생

## 🐛 버그 설명
루트 엔드포인트(`/`) 접근 시 `NameError: name 'undefined_variable' is not defined` 오류가 발생합니다.

## 🔍 재현 방법
1. API 서버 실행
2. `GET /` 엔드포인트 호출
3. 500 Internal Server Error 발생

## 📋 테스트 결과
```
FAILED tests/test_main.py::test_read_root - NameError: name 'undefined_variable' is not defined
```

## 🎯 예상 동작
- 루트 엔드포인트 접근 시 정상적인 JSON 응답 반환
- `{"message": "Chat API Server"}` 형태의 응답 예상

## 🛠️ 해결 방안
`backend/app/main.py`의 root 함수에서 정의되지 않은 변수 `undefined_variable` 참조를 제거해야 합니다.

**오류 코드 위치:**
```python
@app.get("/")
async def root():
    """루트 엔드포인트"""
    # 의도적 버그: 존재하지 않는 변수 참조
    return {"message": "Chat API Server", "version": undefined_variable}
```

## 📎 추가 정보
- **커밋 해시:** `3192704`
- **영향 범위:** 루트 엔드포인트 (`/`)
- **심각도:** High (API 기본 기능 장애)

--- a/backend/app/main.py
+++ b/backend/app/main.py
@@ -24,8 +24,7 @@ app.include_router(chat.router)
 @app.get("/")
 async def root():
     """루트 엔드포인트"""
-    # 의도적 버그: 존재하지 않는 변수 참조
-    return {"message": "Chat API Server", "version": undefined_variable}
+    return {"message": "Chat API Server"}
 
 
 @app.get("/health")

